rust/agama-autoinstall/src/loader.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) [2025] SUSE LLC
+//
+// All Rights Reserved.
+//
+// This program is free software; you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 2 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, contact SUSE LLC.
+//
+// To contact SUSE LLC about this file by physical or electronic mail, you may
+// find current contact information at www.suse.com.
+
+use std::{io::Write, process::Stdio};
+
+use agama_lib::{
+    http::BaseHTTPClient,
+    questions::{
+        http_client::HTTPClient as QuestionsHTTPClient,
+        model::{GenericQuestion, Question},
+    },
+};
+use anyhow::anyhow;
+
+/// It loads the an Agama configuration.
+///
+/// This struct is responsible for reading the configuration from a given URL and,
+/// if wanted, reporting and asking the user about potential problems.
+///
+/// It relies on Agama's command-line to generate and load the new
+/// configuration. In the future, it could rely directly on Agama libraries
+/// instaed of the command-line.
+pub struct ConfigLoader {
+    /// Questions client to interact with the user.
+    questions: QuestionsHTTPClient,
+}
+
+impl ConfigLoader {
+    /// Builds a new loader.
+    pub fn new(http_client: BaseHTTPClient) -> anyhow::Result<Self> {
+        Ok(Self {
+            questions: QuestionsHTTPClient::new(http_client)?,
+        })
+    }
+
+    /// Loads the configuration from the given URL.
+    ///
+    /// When running in interactive mode, it report errors and ask the user
+    /// to continue.
+    pub async fn load(&self, url: &str, interactive: bool) -> anyhow::Result<()> {
```

Comment:
to be honest I am not that happy about mixing that interactive mode with config loader. I kind of see it on different layer..so for me it makes more sense to
1. have class for config loader
2. have class that use config loader and ask for retry via question.

Maybe even more reusable with three classes where the third one is representation for question, so the class 2 can get question and result and return if it should be retried or not.

Reply:
Well, we could move the responsability to ask for retry to the `AutoLoader` struct which is the one that orchestrates the whole process.